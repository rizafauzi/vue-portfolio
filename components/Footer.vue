<template>
	<div class="footer-wrapper">
		<div v-if="isHome" @click="jumpToStart()" class="to-top">
			<img class="top-img" src="~/static/icon/arrow-line.png" />
		</div>
		<div v-else @click="jumpBack()" class="to-back">
			<h3>Back</h3>
		</div>
		<div class="closing">
			<h3>Designed and Developed by Riza Fauzi</h3>
			<h3>Â© 2021 All Rights Reserved.</h3>
		</div>
	</div>
</template>

<script>
export default {
	props: {
		isHome: {
			type: Boolean,
			default() {
				return false
			}
		}
	},
	methods: {
		jumpBack() {
			if (process.browser) {
				window.history.back();
			}
		},
		jumpToStart() {
			if (process.browser) {
				const element = document.getElementById('header')
				if (element) {
					element.scrollIntoView({ behavior: 'smooth', block: 'start' })
				}
			}
		}
	}
}

	
</script>

<style lang="scss" scoped>
.footer-wrapper {
	width: 100%;
	height: 20vh;
	position: relative;
	background: linear-gradient(to top left,#5026B5, #C968E0, #FD7FB9, #FFBBAB);
	.to-top {
		top: -30px;
		width: 60px;
		height: 60px;
		display: flex;
		position: absolute;
		align-items: center;
		border-radius: 100px;
		justify-content: center;
		right: calc(50% - 25px);
		backdrop-filter: blur(20px);
		background: rgba(255, 255, 255, 0.3);
		border: solid 1px rgba(255, 255, 255, 0.8);
		.top-img {
			width: 60%;
			height: 60%;
			object-fit: contain;
			transform: rotate(-90deg);
		}
	}
	.to-back {
		top: -25px;
		display: flex;
		padding: 10px 30px;
		position: absolute;
		align-items: center;
		border-radius: 100px;
		justify-content: center;
		right: calc(50% - 30px);
		backdrop-filter: blur(20px);
		background: rgba(255, 255, 255, 0.3);
		border: solid 1px rgba(255, 255, 255, 0.8);
	}
	.closing {
		width: 100%;
		bottom: 3vh;
		display: flex;
		flex-direction: column;
		justify-content: flex-end;
		align-items: center;
		padding: 20px;
		position: absolute;
	}
}

h3 {
	@media (max-width: 768px) {
		font-size: 12px;
	}
}

.to-top:hover {
	cursor: pointer;
}

.to-back:hover {
	cursor: pointer;
}

</style>
