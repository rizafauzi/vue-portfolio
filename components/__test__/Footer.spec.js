import { shallowMount } from '@vue/test-utils'
import Footer from '../Footer.vue'

describe('Footer.vue', () => {
	let wrapper = null

	beforeEach(() => {
		wrapper = shallowMount(Footer,{
			propsData : 
			})
	})
	afterEach(() => {
		wrapper.destroy()
	})

	it('should render correctly', () => {
		
		expect(wrapper.element).toMatchSnapshot()
	})

	// it('should change text label when button clicked', () => {
	// wrapper.find('.btn-action').trigger('click.prevent');

	// expect(wrapper.vm.label).toBe('Hello');
	// })
})
