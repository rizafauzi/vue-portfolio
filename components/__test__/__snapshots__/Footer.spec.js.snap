// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer.vue should render correctly 1`] = `
<div
  class="container"
>
  <div
    class="section"
  >
    <div
      class="tagline"
    >
      <div
        class="logo"
      >
        <img
          alt="Playfest"
          class="playfest-logo"
          src="~/static/playfest.png"
        />
      </div>
       
      <div
        class="desc"
      >
        <span>
          Stay Updated
        </span>
      </div>
       
      <div
        class="narasi-logo"
      >
        <img
          alt="Narasi"
          src="~/static/logo-yellow.png"
        />
      </div>
    </div>
     
    <div
      class="tagline-mobile"
    >
      <div
        class="narasi-logo"
      >
        <img
          alt="Narasi"
          src="~/static/logo-yellow.png"
        />
      </div>
       
      <div
        class="logo"
      >
        <img
          alt="Playfest"
          class="playfest-logo"
          src="~/static/playfest.png"
        />
      </div>
       
      <div
        class="desc"
      >
        <span>
          Stay Updated
        </span>
      </div>
    </div>
     
    <form>
      <input
        type="email"
      />
       
      <button>
        Subscribe
      </button>
    </form>
     
    <div
      class="social-wrapper"
    >
      <a
        href=""
        target="_blank"
      >
        <svg
          fill="none"
          height="23"
          viewBox="0 0 24 23"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.8124 5.97342H8.51304C7.18834 5.97342 6.1106 7.02164 6.1106 8.31004V16.382C6.1106 17.6704 7.18834 18.7189 8.51304 18.7189H16.8124C18.1371 18.7189 19.2151 17.6707 19.2151 16.382V8.31004C19.2151 7.02164 18.1374 5.97342 16.8124 5.97342ZM12.6627 16.5419C10.2842 16.5419 8.34892 14.6597 8.34892 12.346C8.34892 10.0326 10.2842 8.15042 12.6627 8.15042C15.0416 8.15042 16.9768 10.0326 16.9768 12.346C16.9768 14.6594 15.0413 16.5419 12.6627 16.5419ZM17.1155 9.01791C16.5524 9.01791 16.0945 8.57255 16.0945 8.02521C16.0945 7.47786 16.5524 7.0325 17.1155 7.0325C17.6782 7.0325 18.1361 7.47786 18.1361 8.02521C18.1361 8.57255 17.6782 9.01791 17.1155 9.01791Z"
            fill="#4B2D69"
          />
           
          <path
            d="M12.6628 9.92346C11.2897 9.92346 10.1716 11.0103 10.1716 12.3458C10.1716 13.6818 11.2897 14.769 12.6628 14.769C14.0365 14.769 15.1536 13.6818 15.1536 12.3458C15.1536 11.0106 14.0362 9.92346 12.6628 9.92346Z"
            fill="#4B2D69"
          />
        </svg>
      </a>
       
      <a
        href=""
        target="_blank"
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M14.902 12.2558H12.8422V19.9658H9.72155V12.2558H8.23735V9.54625H9.72155V7.79284C9.72155 6.53897 10.3045 4.57553 12.87 4.57553L15.1816 4.58541V7.21552H13.5044C13.2293 7.21552 12.8424 7.35596 12.8424 7.9541V9.54877H15.1746L14.902 12.2558Z"
            fill="#4B2D69"
          />
        </svg>
      </a>
       
      <a
        href=""
        target="_blank"
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.5274 8.47591C19.0328 8.70003 18.5007 8.85161 17.9426 8.91931C18.5124 8.57052 18.9496 8.01864 19.156 7.35996C18.6229 7.6831 18.0321 7.91752 17.4039 8.04387C16.9006 7.4962 16.1835 7.15393 15.3895 7.15393C13.866 7.15393 12.6304 8.41642 12.6304 9.97302C12.6304 10.194 12.6549 10.4091 12.7022 10.6155C10.4091 10.498 8.37596 9.37574 7.01503 7.66965C6.77758 8.08592 6.64136 8.57052 6.64136 9.08708C6.64136 10.0649 7.12882 10.9281 7.86875 11.4336C7.41668 11.419 6.99096 11.2923 6.61914 11.0806C6.61893 11.0925 6.61893 11.1045 6.61893 11.1163C6.61893 12.4822 7.57039 13.6215 8.83235 13.8803C8.60107 13.9451 8.35683 13.9793 8.10559 13.9793C7.9274 13.9793 7.75476 13.9619 7.58644 13.9291C7.93748 15.0488 8.95622 15.8639 10.1639 15.8868C9.2194 16.6431 8.02987 17.0936 6.73684 17.0936C6.51461 17.0936 6.29444 17.0804 6.079 17.0541C7.2994 17.8542 8.75005 18.3208 10.3081 18.3208C15.3831 18.3208 18.1585 14.0252 18.1585 10.2995C18.1585 10.1774 18.1558 10.0556 18.1505 9.93497C18.69 9.53762 19.1577 9.04125 19.5274 8.47591Z"
            fill="#4B2D69"
          />
        </svg>
      </a>
    </div>
     
    <div
      class="copyright"
    >
      <p>
        Â© 2020 Playfest. Proudly created by www.narasi.tv
      </p>
    </div>
  </div>
</div>
`;
